### PR TITLE
User-facing docs for secret disclosure API

### PR DESCRIPTION
**Pull request by di**
Closes https://github.com/pypi/warehouse/issues/17217.

### CODE REVIEW

**Comment by di** [comment]
New doc pages are:

* User-facing: https://docspypiorg--17236.org.readthedocs.build/api/secrets/
* Dev-facing: https://warehouse--17236.org.readthedocs.build/development/token-scanning.html

File: docs/user/api/secrets.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+The response from a GET request to this URL should return a JSON document with
+the following example structure:
+
+```json
+{
+  "public_keys": [
+    {
+      "key_identifier": "90a421169f0a406205f1563a953312f0be898d3c7b6c06b681aa86a874555f4a",
+      "key": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE9MJJHnMfn2+H4xL4YaPDA4RpJqUq\nkCmRCBnYERxZanmcpzQSXs1X/AljlKkbJ8qpVIW4clayyef9gWhFbNHWAA==\n-----END PUBLIC KEY-----\n",
+      "is_current": false
```
Comment (woodruffw) [inline docs/user/api/secrets.md:62]: Not sure if this is worth mentioning or now, but flagging: the current `_check_signature` helper assumes that the signature is an ECDSA sig, and that the digest is SHA-256. 

As such, integrating parties should be prepared to provide P-256/384/521 keys, and use SHA-256 only (not SHA-384 or SHA-512, despite those being common with P-384 and P-521 respectively).

Ref: https://github.com/pypi/warehouse/blob/7ce8cee7a09c03b0f252e4a0c7da0787ab08bcdb/warehouse/integrations/__init__.py#L122

**Review by woodruffw — APPROVED** [review]
LGTM! Left one comment about the crypto primitives.

File: docs/user/api/secrets.md (hunk 1 above)
Comment (di) [reply docs/user/api/secrets.md:69]: Added in f119687, PTAL.

**Review by woodruffw — APPROVED** [review]
LGTM!

**Pull request merged**